src/p3k/Micropub/Request.php

```diff
@@ -182,6 +182,8 @@ public function toMf2() {
 
   public function __get($k) {
     switch($k) {
+      case 'type':
+        return reset($this->_type);
```

Comment:
What's the reason for `reset` here? Also should this return a string value or array?

Reply:
I was confused by that initially. `_type` appears to always be an array. With JSON input, it comes in as an array, and the form-encoded processing makes it an array at:

https://github.com/aaronpk/p3k-micropub/blob/c19452e5ec64eee2891e6821c93ada7734f82462/src/p3k/Micropub/Request.php#L127)

So the reset makes it return a string.

Reply:
I see... can you do this instead?

```suggestion
        return $this->_type[0];
```

And then also on [line 64](https://github.com/aaronpk/p3k-micropub/blob/master/src/p3k/Micropub/Request.php#L64) add another check that makes sure there is at least one value for type in the request?